File: src/io/github/apollozhu/lottery/lottery/LotteryRollingPanel.java
package io.github.apollozhu.lottery.lottery;

import io.github.apollozhu.lottery.utils.RandomSelector;

import javax.swing.*;

public class LotteryRollingPanel extends LotteryCenterPanel {
    private Timer timer = new Timer(30, ignored -> {
        label.setText(RandomSelector.INSTANCE.getList().get(RandomSelector.INSTANCE.nextIndex()));
        System.out.println(RandomSelector.INSTANCE.nextIndex());
    });

    public LotteryRollingPanel() {
        timer.start();
    }

    @Override
    public void setVisible(boolean aFlag) {
        super.setVisible(aFlag);
        if (aFlag) {
            timer.start();
        } else {
            timer.stop();
        }
    }
}

File: src/io/github/apollozhu/lottery/lottery/LotteryRollingPanel.kt
package io.github.apollozhu.lottery.lottery

import io.github.apollozhu.lottery.utils.RandomSelector
import javax.swing.Timer

class LotteryRollingPanel : LotteryCenterPanel() {
    private val timer = Timer(30) {
        label.text = RandomSelector.list[RandomSelector.nextIndex()]
        checkState()
    }

    fun stop() = timer.stop()

    fun start() {
        if (RandomSelector.hasNext()) timer.start()
    }

    private fun checkState() {
        if (!RandomSelector.hasNext()) stop()
    }

    init {
        start()
    }

    override fun setVisible(aFlag: Boolean) {
        super.setVisible(aFlag)
        if (aFlag) timer.start() else timer.stop()
    }
}
